hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1106,6 +1156,38 @@ private void finishActiveMasterInitialization(MonitoredTask status)
     }
   }
 
+  private void createMissingCFsInMetaDuringUpgrade(
+      TableDescriptor metaDescriptor) throws IOException {
+    TableDescriptor newMetaDesc =
+        TableDescriptorBuilder.newBuilder(metaDescriptor)
+            .setColumnFamily(FSTableDescriptors.getTableFamilyDesc(conf))
+            .setColumnFamily(FSTableDescriptors.getReplBarrierFamilyDesc())
+            .build();
+    long pid = this.modifyTable(TableName.META_TABLE_NAME, () -> newMetaDesc,
```

Comment:
This procedure could be done without ClusterSchemaService?

Reply:
Yes, apparently this can be done without ClusterSchemaService as per my dev testing.